src/GovUk.Education.ExploreEducationStatistics.Admin/Services/Public.Data/DataSetService.cs

```diff
@@ -101,11 +125,89 @@ private static DataSetViewModel MapDataSet(DataSet dataSet)
+
+        return new DataSetSummaryViewModel
+        {
+            Id = dataSet.Id,
+            Title = dataSet.Title,
+            Summary = dataSet.Summary,
+            Status = dataSet.Status,
+            DraftVersion = draftVersion,
+            LatestLiveVersion = latestLiveVersion,
+        };
+    }
+
+    private static DataSetVersionSummaryViewModel MapDataSetVersion(DataSetVersion dataSetVersion, ReleaseFile releaseFile)
+    {
+        return new DataSetVersionSummaryViewModel
+        {
+            Id = dataSetVersion.Id,
+            Version = dataSetVersion.Version,
+            Status = dataSetVersion.Status,
+            Type = dataSetVersion.VersionType,
+            DataSetFileId = (Guid)releaseFile.File.DataSetFileId!,
+            ReleaseVersion = MapReleaseVersion(releaseFile.ReleaseVersion),
+        };
+    }
+
+    private static IdTitleViewModel MapReleaseVersion(ReleaseVersion releaseVersion)
+    {
+        return new IdTitleViewModel
+        {
+            Id = releaseVersion.Id,
+            Title = releaseVersion.Title,
+        };
+    }
+
     private async Task<Either<ActionResult, Publication>> CheckPublicationExists(Guid publicationId,
         CancellationToken cancellationToken)
     {
         return await contentDbContext.Publications
             .AsNoTracking()
             .FirstOrNotFoundAsync(p => p.Id == publicationId, cancellationToken: cancellationToken);
     }
+
+    private async Task<Either<ActionResult, DataSet>> CheckDataSetExists(
+        Guid dataSetId,
+        CancellationToken cancellationToken)
+    {
+        return await publicDataDbContext.DataSets
+            .AsNoTracking()
+            .Include(ds => ds.LatestDraftVersion)
+            .Include(ds => ds.LatestLiveVersion)
+            .Where(ds => ds.Id == dataSetId)
+            .SingleOrNotFoundAsync(cancellationToken);
+    }
+
+    private async Task<Either<ActionResult, IReadOnlyDictionary<Guid, ReleaseFile>>> GetLatestDataSetVersionFilesByFileId(
+        DataSet dataSet,
+        CancellationToken cancellationToken)
+    {
+        if (dataSet.LatestDraftVersion is null && dataSet.LatestLiveVersion is null)
+        {
+            return new Dictionary<Guid, ReleaseFile>();
+        }
+
+        Expression<Func<ReleaseFile, bool>> predicate =
+            dataSet.LatestDraftVersion is not null && dataSet.LatestLiveVersion is not null
+            ? rf => rf.FileId == dataSet.LatestDraftVersion.CsvFileId || rf.FileId == dataSet.LatestLiveVersion.CsvFileId
+            : dataSet.LatestDraftVersion is not null
+            ? rf => rf.FileId == dataSet.LatestDraftVersion.CsvFileId
+            : rf => rf.FileId == dataSet.LatestLiveVersion!.CsvFileId;
+
+        return await contentDbContext.ReleaseFiles
+            .Where(predicate)
+            .Include(rf => rf.ReleaseVersion)
+            .Include(rf => rf.File)
+            .ToDictionaryAsync(rf => rf.FileId, cancellationToken);
```

Comment:
I feel like this might be simpler to understand but it's up to you if you want to go with this. Just mentioning it in case you've not already considered it as a possibility.

```csharp
        if (dataSet.LatestDraftVersion is null && dataSet.LatestLiveVersion is null)
        {
            return new Dictionary<Guid, ReleaseFile>();
        }

        var fileIds = new List<Guid>();

        if (dataSet.LatestDraftVersion is not null)
        {
            fileIds.Add(dataSet.LatestDraftVersion.CsvFileId);
        }

        if (dataSet.LatestLiveVersion is not null)
        {
            fileIds.Add(dataSet.LatestLiveVersion.CsvFileId);
        }

        return await contentDbContext.ReleaseFiles
            .Where(rf => fileIds.Contains(rf.FileId))
            .Include(rf => rf.ReleaseVersion)
            .Include(rf => rf.File)
            .ToDictionaryAsync(rf => rf.FileId, cancellationToken);
```

Reply:
I'm happy to go with that! Probably slightly more readable :) thanks for the suggestion